File: common/src/main/java/com/sos/scheduler/engine/common/system/CpuArchitecture.java
package com.sos.scheduler.engine.common.system;

public enum CpuArchitecture {
    x86 ("x86", "Win32"),
    x64 ("x64", "x64");

    private final String officialName;
    private final String visualStudioName;

    CpuArchitecture(String officialName, String visualStudioName) {
        this.officialName = officialName;
        this.visualStudioName = visualStudioName;
    }

    public static CpuArchitecture cpuArchitecture() {
        return System.getProperty("os.arch").equals("amd64")? x64 : x86;
    }

    public final String officialName() {
        return officialName;
    }

    public final String visualStudioName() {
        return visualStudioName;
    }
}

File: js7-base/shared/src/main/scala/js7/base/system/CpuArchitecture.scala
package js7.base.system

final case class CpuArchitecture(officialName: String, visualName: String)

object CpuArchitecture
{
  val x86 = CpuArchitecture("x86", "Win32")
  val x64 = CpuArchitecture("x64", "x64")

  val cpuArchitecture: CpuArchitecture =
    if (System.getProperty("os.arch") == "amd64")
      x64
    else
      x86
}
